src/Datadog.Trace/Tracer.cs

```diff
@@ -415,6 +417,91 @@ internal void StartDiagnosticObserversInternal()
             }
         }
 
+        internal void WriteDiagnosticLog()
+        {
+            try
+            {
+                var stringWriter = new StringWriter();
+
+                using (var writer = new JsonTextWriter(stringWriter))
+                {
+                    writer.WriteStartObject();
+
+                    writer.WritePropertyName("date");
+                    writer.WriteValue(DateTime.Now);
+
+                    writer.WritePropertyName("os_name");
+#if NETSTANDARD
+                    writer.WriteValue(System.Runtime.InteropServices.RuntimeInformation.OSDescription);
+#else
+                    writer.WriteValue(Environment.OSVersion.ToString());
+#endif
```

Comment:
We should probably include OS platform (x64, x86, etc) and process platform. We should probably get this form `FrameworkDescription` to keep it consistent with other places where we report this.

Reply:
Good point, gonna add the process platform. Is there any value in including the OS platform?

Reply:
Hmm, I guess not. Never mind!